src/System.Private.CoreLib/src/System/Threading/ThreadPool.Portable.cs

```diff
@@ -207,11 +206,7 @@ private void SignalUserWaitHandle()
                 if (handleValue != IntPtr.Zero && handleValue != (IntPtr)(-1))
                 {
                     Debug.Assert(handleValue == handle.DangerousGetHandle());
-#if PLATFORM_WINDOWS
-                    Interop.Kernel32.SetEvent(handle);
-#else
-                    WaitSubsystem.SetEvent(handleValue);
-#endif
+                    EventWaitHandle.Set(handle);
```

Comment:
@kouvel Was there a reason for why this was done this way?

Reply:
Historically the reason was that `EventWaitHandle.Set` didn't exist. I added it later.

Reply:
I believe that was it